pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -645,6 +650,45 @@ func parseDeviceAddress(addrString string) []string {
 	return addrs
 }
 
+func expandDiskImages(vmi *v1.VirtualMachineInstance, domain *api.Domain) {
+	logger := log.Log.Object(vmi)
+	for _, disk := range domain.Spec.Devices.Disks {
+		if imageCanBeExpanded(disk) {
+			logger.Infof("pre-start expansion of disk %v", disk)
+			err := expandDiskImage(disk.Source.File, disk.Size)
+			if err != nil {
+				logger.Reason(err).Error("failed to expand disk image")
+			}
+		}
+	}
+}
+
+func imageCanBeExpanded(disk api.Disk) bool {
```

Comment:
Perhaps `diskShouldBeExpanded()` would more accurately describe what this function does?

Reply:
Changed it for shouldExpand(disk)